stream.go

```diff
@@ -321,6 +347,53 @@ SEND_CLOSE:
 	return nil
 }
 
+// CloseWrite is used to close this side's write end of the stream.
+func (s *Stream) CloseWrite() error {
+	s.stateLock.Lock()
+	s.flags |= writeCloseFlag
+	switch s.state {
+	// Opened means we need to signal a close
+	case streamSYNSent:
+		fallthrough
+	case streamSYNReceived:
+		fallthrough
+	case streamEstablished:
+		goto SEND_CLOSE
+
+	case streamLocalClose:
```

Comment:
I think you mean that seeing `streamLocalClose` means we do nothing and continue after the switch but Go makes it hard to here if that's the case or if it's a programming error. Could you add a comment to the bottom saying something like "ok, handle normally" or something? Thanks!